### PR TITLE
VIT-2513 BLE Glucose Service reader improvements

### PR DESCRIPTION
**Pull request by andersio**
Improve the BLE Glucose Service (1808) reader in VitalDevices:

1. Fixed unit handling: Now handles both kg/L and mol/L correctly as per BLE spec.

2. Fixed binary payload decoding: Respect flags and optional frames; handle the sfloat format correctly.

3. The `GlucoseMeter.read(device:)` publisher now either completes with one batch of all stored records, or error out. This is new change from the current behaviour (never ending publisher, with samples in chunks). 

    * This is achieved by listening to the RACP response indication (notification), which must be issued by a compliant device after they have finished sending out all the measurements via BLE notifications, or when an error is encountered.

4. Added device support for Vital BLE Simulator.

Note: Blood Pressure service to be improved separately

### CODE REVIEW

File: Sources/VitalDevices/DeviceReading/BloodPressureReader1810.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ class BloodPressureReader1810: BloodPressureReadable {
   
   init(manager: CentralManager = .live(), queue: DispatchQueue) {
     self.manager = manager
-    self.queue = queue
+    self.queue = DispatchQueue(label: "co.tryvital.VitalDevices.BloodPressureReader1810", target: queue)
```
Comment (RuiAAPeres) [inline Sources/VitalDevices/DeviceReading/BloodPressureReader1810.swift:19]: co vs io?

File: Sources/VitalDevices/DeviceReading/GlucoseMeter1808.swift [hunk 2]
```diff
@@ -103,21 +140,28 @@ class GlucoseMeter1808: GlucoseMeterReadable {
 }
 
 private func toGlucoseReading(data: Data?) -> QuantitySample? {
-  guard let data = data else {
+  guard let data = data, data.count >= 10 else {
```
Comment (RuiAAPeres) [inline Sources/VitalDevices/DeviceReading/GlucoseMeter1808.swift:143]: why `data.count >= 10`?

**Review by RuiAAPeres — APPROVED** [review]

File: Sources/VitalDevices/DeviceReading/GlucoseMeter1808.swift (hunk 2 above)
Comment (andersio) [reply Sources/VitalDevices/DeviceReading/GlucoseMeter1808.swift:143]: All the mandatory fields of a glucose measurement record add up to 10 bytes

File: Sources/VitalDevices/DeviceReading/BloodPressureReader1810.swift (hunk 1 above)
Comment (andersio) [reply Sources/VitalDevices/DeviceReading/BloodPressureReader1810.swift:19]: oops

**Review by RuiAAPeres — APPROVED** [review]

**Pull request merged**